src/internal/operations/backup_integration_test.go

```diff
@@ -191,67 +206,210 @@ func checkMetadataFilesExist(
+// feels like something we can clean up.  But, it's not a strong need, so
+// this gets to stay for now.
+// ---------------------------------------------------------------------------
 
-	for _, fName := range files {
-		p, err := path.Builder{}.
-			Append(fName).
-			ToServiceCategoryMetadataPath(tenant, user, service, category, true)
-		if !assert.NoError(t, err, "bad metadata path") {
-			continue
-		}
+// the params here are what generateContainerOfItems passes into the func.
+// the callback provider can use them, or not, as wanted.
+type dataBuilderFunc func(id, now, subject, body string) []byte
 
-		dir, err := p.Dir()
-		if !assert.NoError(t, err, "parent path") {
-			continue
-		}
+//revive:disable:context-as-argument
+func generateContainerOfItems(
+	t *testing.T,
+	ctx context.Context,
+	gc *connector.GraphConnector,
+	service path.ServiceType,
+	cat path.CategoryType,
+	sel selectors.Selector,
+	tenantID, userID, destFldr string,
+	howMany int,
+	dbf dataBuilderFunc,
+) (*details.Details, error) {
+	//revive:enable:context-as-argument
+	t.Helper()
+
+	items := make([]incrementalItem, 0, howMany)
 
-		paths = append(paths, p)
-		pathsByRef[dir.ShortRef()] = append(pathsByRef[dir.ShortRef()], fName)
+	for i := 0; i < howMany; i++ {
+		var (
+			now       = common.Now()
+			nowLegacy = common.FormatLegacyTime(time.Now())
+			id        = uuid.NewString()
+			subject   = "incr_test " + now[:16] + " - " + id[:8]
+			body      = "incr_test " + cat.String() + " generation for " + userID + " at " + now + " - " + id
+		)
+
+		items = append(items, incrementalItem{
+			name: id,
+			data: dbf(id, nowLegacy, subject, body),
+		})
 	}
 
-	cols, err := kw.RestoreMultipleItems(ctx, bup.SnapshotID, paths, nil)
-	assert.NoError(t, err)
+	collections := []incrementalCollection{{
+		pathElements: []string{destFldr},
+		category:     cat,
+		items:        items,
+	}}
 
-	for _, col := range cols {
-		itemNames := []string{}
+	dest := control.DefaultRestoreDestination(common.SimpleTimeTesting)
+	dest.ContainerName = destFldr
 
-		for item := range col.Items() {
-			assert.Implements(t, (*data.StreamSize)(nil), item)
+	dataColls, err := buildCollections(
+		t,
+		service,
+		tenantID, userID,
+		dest,
+		collections)
+	if err != nil {
+		return nil, err
+	}
 
-			s := item.(data.StreamSize)
-			assert.Greaterf(
-				t,
-				s.Size(),
-				int64(0),
-				"empty metadata file: %s/%s",
-				col.FullPath(),
-				item.UUID(),
-			)
+	return gc.RestoreDataCollections(ctx, sel, dest, dataColls)
+}
 
-			itemNames = append(itemNames, item.UUID())
-		}
+type incrementalItem struct {
+	name string
+	data []byte
+}
 
-		assert.ElementsMatchf(
+type incrementalCollection struct {
+	// Elements (in order) for the path representing this collection. Should
+	// only contain elements after the prefix that corso uses for the path. For
+	// example, a collection for the Inbox folder in exchange mail would just be
+	// "Inbox".
```

Comment:
isn't this just the `path.folders()` call? 

https://github.com/alcionai/corso/blob/20ec708ea357c26afe0076e31f05eacb8597a688/src/pkg/path/resource_path.go#L206